### ISSUE TITLE
Takeover mode cannot find TypeScript at path defined in settings.json anymore

### ISSUE THREAD

**Issue by Anoesj** [vuejs/language-tools#2637]
Ever since I updated to vue-tsc 1.4.x, it seems like Volar cannot find my TypeScript lib anymore in Takeover mode. I defined this in `.vscode/settings.json`, because my folder structure is a little different (no `node_modules` in the root directory):
```json
{
  "typescript.tsdk": "SomeDir/AnotherDir/node_modules/typescript/lib"
}
```

I get `x.x.x (takover)` in the bottom bar in VSCode and when clicked, it shows:
![Code_10Wdmgr2DN](https://user-images.githubusercontent.com/7874160/233493580-77c24acc-2600-48fc-85c8-5a818db22804.png)

Interestingly, selecting VSCode's version of TS doesn't work anymore either, it shows notifications like "Vue Semantic Server client: couldn't create connection to server" outputs this to the Vue Semantic Server output tab:
```
[Info  - 11:40:58 PM] Connection to server got closed. Server will restart.
true
[Error - 11:40:58 PM] Server initialization failed.
  Message: Request initialize failed with message: Can't find typescript.js or tsserverlibrary.js in undefined
  Code: -32603 
[Error - 11:40:58 PM] Vue Semantic Server client: couldn't create connection to server.
  Message: Request initialize failed with message: Can't find typescript.js or tsserverlibrary.js in undefined
  Code: -32603 
[Error - 11:40:58 PM] Restarting server failed
  Message: Request initialize failed with message: Can't find typescript.js or tsserverlibrary.js in undefined
  Code: -32603 
[Error - 11:41:00 PM] The Vue Semantic Server server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.
```

Any chance the vue-tsc update has anything to do with this?

**Comment by johnsoncodehk** [comment]
Can you try relative path?

```json
{
  "typescript.tsdk": "./SomeDir/AnotherDir/node_modules/typescript/lib"
}
```

**Comment by wenyongda** [comment]
I have the same problem.
{
   "typescript.tsdk":  "node_modules/typescript/lib"
}
![image](https://user-images.githubusercontent.com/68416083/233518458-f6f81e1e-4628-46c4-8e73-673b01027ce9.png)
![image](https://user-images.githubusercontent.com/68416083/233518482-cc490f3a-9236-4993-8561-75c4bf5a3d2a.png)
```
[Error - 09:18:39] Restarting server failed
  Message: Request initialize failed with message: Can't find typescript.js or tsserverlibrary.js in undefined
  Code: -32603 
[Error - 09:18:42] The Vue Semantic Server server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.
[Error - 09:18:54] Server initialization failed.
  Message: Request initialize failed with message: Can't find typescript.js or tsserverlibrary.js in undefined
  Code: -32603 
[Error - 09:18:54] Vue Semantic Server client: couldn't create connection to server.
  Message: Request initialize failed with message: Can't find typescript.js or tsserverlibrary.js in undefined
  Code: -32603 
[Error - 09:18:56] The Vue Semantic Server server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.
```

**Comment by Tanimodori** [comment]
1.4.0 breaks tsdk for me too. It cannot loads correct in my rush monorepo. Tried to set `typescript.tsdk` to the following values but none of them works.

* `common/temp/node_modules/typescript/lib`
* `./common/temp/node_modules/typescript/lib`

Rolling back to 1.3.x won't solve this problem for me (tried 1.3.19, 1.3.16).

**Comment by johnsoncodehk** [comment]
Please try v1.4.1.

**Comment by wenyongda** [comment]
It's normal.Thank you very much

**Comment by Anoesj** [comment]
I can confirm it's working again, thanks for the super fast response @johnsoncodehk :heart:

**Comment by Tanimodori** [comment]
It works for me in 1.4.1. Thanks for the quick fix! ❤️

**Comment by 0biWanKenobi** [comment]
Sorry but it's still broken for me, I have x.x.x (takeover) in the bottom toolbar and this error when opening the TS version picker:
![image](https://user-images.githubusercontent.com/4061104/233572518-60d85f8f-4834-4cd0-9b80-584418826db5.png)

I am on Windows, `vue-tsc` version 0.31.4

**Comment by rajeshtva** [comment]
Apparently broken on my system as well..
volar version: 1.4.2

```
Request initialize failed with message: Can't find typescript.js or tsserverlibrary.js in /home/rex/.vscode/extensions/ms-vscode.vscode-typescript-next-5.1.20230421/extensions/node_modules/typescript/lib
```

```
[Error - 12:04:51] Server initialization failed.
  Message: Request initialize failed with message: Can't find typescript.js or tsserverlibrary.js in /home/rex/.vscode/extensions/ms-vscode.vscode-typescript-next-5.1.20230421/extensions/node_modules/typescript/lib
  Code: -32603 
[Error - 12:04:51] Vue Semantic Server client: couldn't create connection to server.
  Message: Request initialize failed with message: Can't find typescript.js or tsserverlibrary.js in /home/rex/.vscode/extensions/ms-vscode.vscode-typescript-next-5.1.20230421/extensions/node_modules/typescript/lib
  Code: -32603 

```

**Comment by Eyre-S** [comment]
> Apparently broken on my system as well.. volar version: 1.4.2
> 
> ```
> Request initialize failed with message: Can't find typescript.js or tsserverlibrary.js in /home/rex/.vscode/extensions/ms-vscode.vscode-typescript-next-5.1.20230421/extensions/node_modules/typescript/lib
> ```
> 
> ```
> [Error - 12:04:51] Server initialization failed.
>   Message: Request initialize failed with message: Can't find typescript.js or tsserverlibrary.js in /home/rex/.vscode/extensions/ms-vscode.vscode-typescript-next-5.1.20230421/extensions/node_modules/typescript/lib
>   Code: -32603 
> [Error - 12:04:51] Vue Semantic Server client: couldn't create connection to server.
>   Message: Request initialize failed with message: Can't find typescript.js or tsserverlibrary.js in /home/rex/.vscode/extensions/ms-vscode.vscode-typescript-next-5.1.20230421/extensions/node_modules/typescript/lib
>   Code: -32603 
> ```


There's the same problem on my computer.

seems volar trying to find typescript lib on `~/.vscode/extensions/ms-vscode.vscode-typescript-next-5.1.20230421/extensions/node_modules/typescript/lib` but the actuall file location is `~/.vscode/extensions/ms-vscode.vscode-typescript-next-5.1.20230421/node_modules/typescript/lib`.

I solved this problem by rolling back my volar version from 1.4.2 to 1.4.1.

**Comment by H0nGzA1** [comment]
<img width="1090" alt="image" src="https://user-images.githubusercontent.com/21973212/233777639-d6a4d762-7981-4655-a276-b391d716d8ea.png">

<img width="1113" alt="image" src="https://user-images.githubusercontent.com/21973212/233777652-4c07120b-1f4b-49d8-9e52-368ae58fdd4e.png">
for me select this it's working

**Comment by yingpwu** [comment]
Change the extension of Vue Language Features (Volar) and  TypeScript Vue Plugin (Volar)  will backto work ,
This problem confused me 2 hours ,mygod......
I turn off the autoupdate now

**Comment by johnsoncodehk** [comment]
The correct issue is #2663, it should be fixed and released in v1.4.3, please don't reply with an obsolete issue.
